### DISCUSSION TITLE
有人知道什么方法能获取defineConfig的配置?

### DISCUSSION THREAD

**Discussion by hanjeahwan** [umijs/umi#7534] [Q&A]
比如说我想在react component以外的地方获取配置
我尝试这样写拿不到 有哪位知道如何吗 
```
import config from '../../../config/config';

export function anything() {
  return config.anything
}
```

**Comment by SilentFlute** [comment]
我自己试了下获取配置, 如果使用了`defineConfig`就无法获取了, 会报错, 具体原因不是太了解, 可能是这个函数只在`node`环境中有
因此在写配置的时候不用这个函数就能获取到了, 伪代码如下:
`/config/config.ts`:
```
export default {
  //...
};
```
其他地方引入即可:
`x.ts:`:
```
import cfg from '/{path}/config.ts';
```
关于`defineConfig`的作用, 它的`d.ts`中有如下说明:
```
import { IConfigFromPlugins } from '@@/core/pluginConfig';
import type { IConfig } from '@umijs/preset-umi';
declare type ConfigType = IConfigFromPlugins & IConfig;
/**
 * 通过方法的方式配置umi，能带来更好的 typescript 体验
 * @param  {ConfigType} config
 * @returns ConfigType
 */
export declare function defineConfig(config: ConfigType): ConfigType;
export {};
```

**Comment by xiaohuoni** [comment]
因为 “配置” 是给 node 侧使用的，他不会包含在浏览器侧。
最主要报错的原因是，在浏览器侧，umi 是通过别名 'umi: "@@/exports",' 提供的，
所以你在浏览器侧使用 `import from "umi"` 其实是 `import from "src/.umi/exports.ts"`
而在 config/config（或者 mock 之类的 node 侧） 中 `import from "umi"` 其实是 `import from "node_modules/umi/dist/index.js"`
而 `defineConfig` 不在 `exports` 中，所以无法使用。
你是什么需求需要从配置中取？我觉得最常见的应该是判断环节变量或者定义某个常量，可以通过   [define: { FOO: 'bar' }](https://umijs.org/docs/api/config#define) 配置实现。

**Reply by SilentFlute** [reply]
哦哦, 原来如此, 虎哥, 那如果是需要获取配置中的`title`, 配合菜单的标题, 使用`Helmet`来设置页面的标题呢? 这个配置中的`title`在代码中该如何获取呢?

**Reply by xiaohuoni** [reply]
配置中的 title 本身就会设置页面的标题吧？现在关于 title 是有点分离的，layout 配置中也有一个 title。估计要做动态 title 的话，可以先去现在页面的 title，然后再操作？

**Reply by SilentFlute** [reply]
先取当前页面的title然后再操作, 也是个方法哈

**Comment by fz6m** [comment]
把需要复用的配置抽取出来，保证这个文件是纯的，没有依赖，然后就可以在 client（项目里） 或 node（配置文件）同时 import 获取到了。

**Reply by SilentFlute** [reply]
哦哦, 也对, 这也是个方法
